File: packages/nerva-py/nerva/policy/noop.py
"""Noop policy engine — allows everything without recording.

Use during development or testing when policy enforcement is not needed.
"""

from __future__ import annotations

from typing import TYPE_CHECKING

from nerva.policy import ALLOW, PolicyAction, PolicyDecision

if TYPE_CHECKING:
    from nerva.context import ExecContext


class NoopPolicyEngine:
    """Policy engine that permits every action unconditionally.

    No state is tracked and ``record`` is a silent no-op. Satisfies the
    ``PolicyEngine`` protocol so it can be used as a drop-in replacement
    for any real engine.
    """

    async def evaluate(
        self, action: PolicyAction, ctx: ExecContext
    ) -> PolicyDecision:
        """Always returns ``ALLOW``.

        Args:
            action: The action to evaluate (ignored).
            ctx: Execution context (ignored).

        Returns:
            The pre-built ``ALLOW`` decision.
        """
        return ALLOW

    async def record(
        self,
        action: PolicyAction,
        decision: PolicyDecision,
        ctx: ExecContext,
    ) -> None:
        """No-op — nothing to record.

        Args:
            action: The evaluated action (ignored).
            decision: The decision made (ignored).
            ctx: Execution context (ignored).
        """

File: packages/nerva-js/src/policy/noop.ts
/**
 * Noop policy engine — allows everything without recording.
 *
 * Use during development or testing when policy enforcement is not needed.
 *
 * @module policy/noop
 */

import type { ExecContext, PolicyAction, PolicyDecision } from "./index.js";
import { ALLOW } from "./index.js";

// ---------------------------------------------------------------------------
// NoopPolicyEngine
// ---------------------------------------------------------------------------

/**
 * Policy engine that permits every action unconditionally.
 *
 * No state is tracked and `record` is a silent no-op. Satisfies the
 * {@link PolicyEngine} interface so it can be used as a drop-in replacement
 * for any real engine.
 */
export class NoopPolicyEngine {
  /**
   * Always returns {@link ALLOW}.
   *
   * @param _action - The action to evaluate (ignored).
   * @param _ctx - Execution context (ignored).
   * @returns The pre-built ALLOW decision.
   */
  async evaluate(
    _action: PolicyAction,
    _ctx: ExecContext,
  ): Promise<PolicyDecision> {
    return ALLOW;
  }

  /**
   * No-op — nothing to record.
   *
   * @param _action - The evaluated action (ignored).
   * @param _decision - The decision made (ignored).
   * @param _ctx - Execution context (ignored).
   */
  async record(
    _action: PolicyAction,
    _decision: PolicyDecision,
    _ctx: ExecContext,
  ): Promise<void> {
    // Intentionally empty — noop engine does not track decisions.
  }
}
